experiments/sample_ssl4eo.py

```diff
@@ -0,0 +1,178 @@
+#!/usr/bin/env python3
+
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+""" Sample patch locations for downloading
+
+### run the script:
+## sample new locations with rtree overlap search
+python sample_ssl4eo.py \
+    --save-path ./data \
+    --size 1320 \
+    --num-cities 10000 \
+    --std 50 \
+    --indices-range 0 250000
+
+## resume from interruption
+python sample_ssl4eo.py \
+    --save-path ./data \
+    --size 1320 \
+    --num-cities 10000 \
+    --std 50 \
+    --resume ./data/sampled_locations.csv \
+    --indices-range 0 250000
+
+### Notes
+# The script will sample locations with rtree overlap search.
+# The script will save the sampled locations to a csv file.
+# By default, GaussianSampler is used to sample locations with a standard deviation
+ of 50 km from top 10000 populated cities.
+# Size (meter) is half the wanted patch size.
+
+"""
+
+import argparse
+import csv
+import os
+import time
+import warnings
+from typing import Tuple
+
+import numpy as np
+import pandas as pd
+from rtree import index
+from torchvision.datasets.utils import download_and_extract_archive
+from tqdm import tqdm
+
+warnings.simplefilter("ignore", UserWarning)
+
+
+""" samplers to get locations of interest points"""
+
+
+def get_world_cities(
+    download_root: str = "world_cities", size: int = 10000
+) -> pd.DataFrame:
+    url = "https://simplemaps.com/static/data/world-cities/basic/simplemaps_worldcities_basicv1.71.zip"  # noqa: E501
+    filename = "worldcities.csv"
+    if not os.path.exists(os.path.join(download_root, os.path.basename(url))):
+        download_and_extract_archive(url, download_root)
+    cols = ["city", "lat", "lng", "population"]
+    cities = pd.read_csv(os.path.join(download_root, filename), usecols=cols)
+    cities.at[8436, "population"] = 50789  # fix one bug (Tecax) in the csv file
+    cities = cities.sort_values(by=["population"], ascending=False).head(size)
+    return cities
+
+
+def km2deg(kms: float, radius: float = 6371) -> float:
+    return kms / (2.0 * radius * np.pi / 360.0)
+
+
+def sample_point(cities: pd.DataFrame, std: float) -> Tuple[float, float]:
+    city = cities.sample()
+    point = (float(city["lng"]), float(city["lat"]))
+    std = km2deg(std)
+    lon, lat = np.random.normal(loc=point, scale=[std, std])
+    return (lon, lat)
+
+
+def create_bbox(
+    coords: Tuple[float, float], bbox_size_degree: float
+) -> Tuple[float, float, float, float]:
+    lon, lat = coords
+    bbox = (
+        lon - bbox_size_degree,
+        lat - bbox_size_degree,
+        lon + bbox_size_degree,
+        lat + bbox_size_degree,
+    )
+    return bbox
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--save-path", type=str, default="./data/", help="dir to save data"
+    )
+    parser.add_argument(
+        "--size", type=float, default=1320, help="half patch size in meters"
```

Comment:
If I want to download 264 x 264 px image patches, would I use 264 * 30 (For Landsat with 30 m resolution)?

Reply:
landsat 8 has two bands with 15 m resolution, if want to match them, should be 264*15 / 2 = 1980

Reply:
We'll prob match the 30 m bands and downsample the 15 m bands.